Manito/Manito/Screens/CreateRoom/ViewModel/CreateRoomViewModel.swift

```diff
@@ -106,84 +164,91 @@ final class CreateRoomViewModel: BaseViewModelType {
+    init(usecase: CreateRoomUsecase) {
+        self.usecase = usecase
     }
     
     // MARK: - func
     
+    private func cutTextByMaxCount(text: String, maxCount: Int) -> String {
+        let isOverMaxCount = self.isOverMaxCount(titleCount: text.count, maxCount: maxCount)
+        
+        if isOverMaxCount {
+            let endIndex = text.index(text.startIndex, offsetBy: maxCount)
+            let fixedText = text[text.startIndex..<endIndex]
+            return String(fixedText)
+        }
+        return text
+    }
+    
     private func isOverMaxCount(titleCount: Int, maxCount: Int) -> Bool {
-        if titleCount > maxCount { return true }
-        else { return false }
+        return titleCount > maxCount ? true : false
     }
     
-    private func runActionByStep(step: CreateRoomStep) -> CurrentNextStep {
-        switch step {
-        case .chooseCharacter:
-            self.requestCreateRoom(roomInfo: CreatedRoomInfoRequestDTO(title: self.titleSubject.value,
-                                                                       capacity: self.capacitySubject.value,
-                                                                       startDate: self.startDateSubject.value,
-                                                                       endDate: self.endDateSubject.value))
-            return (step, step.next())
+    private func initStep() -> StepButtonState {
+        return (self.currentStep, false)
+    }
+    
+    private func nextStep() -> StepButtonState {
+        switch currentStep {
+        case .capacity:
+            self.currentStep = currentStep.next()
+            return (self.currentStep, self.dateIsEmpty())
+        case .character:
+            let roomInfo = CreateRoomInfo(title: self.titleSubject.value,
+                                          capacity: self.capacitySubject.value,
+                                          startDate: self.startDateSubject.value,
+                                          endDate: self.endDateSubject.value)
+            self.dispatchCreateRoom(roomInfo: roomInfo)
+            return (self.currentStep, true)
         default:
-            return (step, step.next())
+            self.currentStep = currentStep.next()
+            return (self.currentStep, true)
         }
     }
     
-    private func previous(step: CreateRoomStep) -> CreateRoomStep {
-        switch step {
-        case .inputTitle:
-            return .inputTitle
-        case .inputCapacity:
-            return .inputTitle
-        case .inputDate:
-            return .inputCapacity
-        case .checkRoom:
-            return .inputDate
-        case .chooseCharacter:
-            return .checkRoom
-        }
+    private func previousStep() -> StepButtonState {
+        self.currentStep = currentStep.previous()
+        return (self.currentStep, true)
+    }
+    
+    private func dateIsEmpty() -> Bool {
+        return self.endDateSubject.value.isEmpty ? false : true
```

Comment:
p4;

여기도 수정할 순 있는데, 느낌표가 너무 앞에 붙어서 조오금 그렇네요...ㅎㅎ
```suggestion
        return !self.endDateSubject.value.isEmpty
```

Reply:
네네 `!` 부호를 통해 Bool 값을 반대로 전달해주면 삼항연산자를 사용하지 않을 수 있어서 남겨봤습니다..!

Reply:
그렇네요 제가 완전히 딴걸 생각하고 있었어요!! 감사합니다 !!

[3074c6e](https://github.com/DeveloperAcademy-POSTECH/MC2-Team5-Firefighter/pull/589/commits/3074c6ec208c5e5a1333ce988509209fc22568da)